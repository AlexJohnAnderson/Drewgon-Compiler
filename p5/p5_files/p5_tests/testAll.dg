int a;
void func(){
    int y;
}

int ifTest(){
    if(a)
    {
        return 10;
    }
}

int forTest(){
    for(int i; 10; i++)
    {
        return 10;
    }
}

void whileTest(){
    int a;
    a = -12;
    while(func())
    {
        return;
    }
}
void main(){
    if(func())
    {
        return;
    }
    output(main());
}



