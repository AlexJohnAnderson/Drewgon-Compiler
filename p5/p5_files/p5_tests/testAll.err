FATAL [7,8]-[7,9]: Non-bool expression used as an if condition
FATAL [14,16]-[14,18]: Non-bool expression used as a loop condition
FATAL [23,11]-[23,17]: Non-bool expression used as a loop condition
FATAL [29,8]-[29,14]: Non-bool expression used as an if condition
FATAL [33,5]-[33,18]: Attempt to output void
Type Analysis Failed
