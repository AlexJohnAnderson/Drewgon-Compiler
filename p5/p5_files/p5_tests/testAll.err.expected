FATAL [7,8]-[7,9]: Non-bool expression used as an if condition
FATAL [14,16]-[14,18]: Non-bool expression used as a loop condition
FATAL [23,11]-[23,17]: Non-bool expression used as a loop condition
FATAL [25,16]-[25,20]: Return with a value in void function
FATAL [29,8]-[29,14]: Non-bool expression used as an if condition
FATAL [31,16]-[31,18]: Return with a value in void function
FATAL [33,12]-[33,18]: Attempt to output void
Type Analysis Failed
